### PR TITLE
config: allow whitespaces in config path

### PR DESCRIPTION
**Pull request by columbarius**
Backport configpath building from xdg-desktop-portal-wlr.

### CODE REVIEW

File: sway/config.c [hunk 1]
```diff
@@ -338,34 +338,62 @@ static bool file_exists(const char *path) {
...
-			if (file_exists(path)) {
-				return path;
-			}
-			free(path);
+	struct config_path {
+		const char *prefix;
+		const char *config_folder;
+	};
+
+	static struct config_path config_paths[6];
```
Comment (emersion) [inline sway/config.c:370]: Hm, this is getting a little out of hand. Maybe can be simplified to:

```c
struct config_path config_paths[] = {
    { .prefix = "asdf", .config_folder = "asdf" },
    …
};
```

Comment (columbarius) [reply sway/config.c:370]: I would like to do that, but that does only work for constant strings and not for variables.

Comment (columbarius) [reply sway/config.c]: Put everything that is constant into the struct and added the variable bits afterwards. I think that's the best we can do, except having one array for all paths and one for all folders and a strange function which maps the order of the config files to the place in each array. But that will probably not make things clearer.

Also change it to calculate the number of tried config files at runtime.

Comment (emersion) [reply sway/config.c]: Maybe try dropping the `const` qualifier for the array

Comment (columbarius) [reply sway/config.c]: You mean the static qualifier? That solved the issue. Didn't know static has some form of constant global behaviour.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Comment by columbarius** [comment]
Thanks for merging!

**Pull request merged**